src/context/PeopleContext.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { createContext, useCallback, useMemo, useState } from 'react';
+import { Person } from '../types';
+import { useSearchParams } from 'react-router-dom';
+import { SortTypeValue } from '../types/SortTypeValue';
+import { ContextType, PropsPeopleContext } from '../types/ContextType';
+
+export const Context = createContext<ContextType>({
+  loading: false,
+  setLoading: () => {},
+  error: false,
+  setError: () => {},
+  people: [],
+  setPeople: () => {},
+  visiblePeople: [],
+});
+
+export const PeopleContext: React.FC<PropsPeopleContext> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || null;
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const sorted = useCallback(
+    (sortVal: SortTypeValue, arr: Person[]) => {
+      return [...arr].sort((a: Person, b: Person) => {
+        switch (sortVal) {
+          case 'name':
+            return order === 'desc'
+              ? b.name.localeCompare(a.name)
+              : a.name.localeCompare(b.name);
+          case 'sex':
+            return order === 'desc'
+              ? b.sex.localeCompare(a.sex)
+              : a.sex.localeCompare(b.sex);
```

Comment:
```suggestion
          case 'name':
          case 'sex':
            return order === 'desc'
              ? b[sortVal].localeCompare(a[sortVal])
              : a[sortVal].localeCompare(b[sortVal]);
```
you can combine some cases